Core/Main/elastixlib.cxx

```diff
@@ -46,9 +46,9 @@ namespace elastix
  * ******************* Constructor ***********************
  */
 
-ELASTIX::ELASTIX() :
-  m_ResultImage( 0 )
+ELASTIX::ELASTIX()
 {
+  m_ResultImage = nullptr;
```

Comment:
I prefer to have the `nullptr` exactly there where it had the `0`: in the mem-initializer-list of the `ELASTIX()` constructor. With your commit,  `ELASTIX()` will first default-construct `m_ResultImage`, and then assign `nullptr` to  `m_ResultImage`. The original code directly constructed `m_ResultImage` with `0`, without an extra assignment.

Reply:
Yes I wanted to talk to you about this. SmartPointer has a constructor that takes a nullptr, so if you write m_ResultImage(nullptr) the object will get initialized so checks like if(m_ResultImage) will be true whereas we wanted it just be a nullptr so if(m_ResultImage) would be false as before (when initialized to zero). If we used if(m_ResultImage.IsNull()) everywhere in the codebase instead of if(m_ResultImage) that be fine but this is a large task to check and beyond the scope of this PR. Is there a way to initialize m_ResultImage to nullptr in the member initializer list when it has a constructor that takes a nullptr?

Reply:
Sure, if you do `m_ResultImage(nullptr)` in the mem-initializer-list, it will call `SmartPointer (std::nullptr_t p)` from https://github.com/InsightSoftwareConsortium/ITK/blob/v5.0rc01/Modules/Core/Common/include/itkSmartPointer.h#L69

`if(m_ResultImage)` calls the function `SmartPointer::operator bool()` from https://github.com/InsightSoftwareConsortium/ITK/blob/v5.0rc01/Modules/Core/Common/include/itkSmartPointer.h#L113 which returns false when it is a `nullptr`.

`if(m_ResultImage)` is equivalent to `if( ! m_ResultImage.IsNull() )`.
It is also equivalent to `if ( m_ResultImage != nullptr )`.
I prefer the last syntax, but that's just my personal taste. I would not change it right now.

Does that answer your question?